src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -753,17 +746,32 @@ uct_ib_mlx5_devx_check_odp(uct_ib_mlx5_md_t *md,
         return status;
     }
 
-    if (!UCT_IB_MLX5DV_GET(odp_cap, odp, ud_odp_caps.send) ||
-        !UCT_IB_MLX5DV_GET(odp_cap, odp, rc_odp_caps.send) ||
-        !UCT_IB_MLX5DV_GET(odp_cap, odp, rc_odp_caps.write) ||
-        !UCT_IB_MLX5DV_GET(odp_cap, odp, rc_odp_caps.read)) {
+    if (UCT_IB_MLX5DV_GET(query_hca_cap_out, out,
+                          capability.odp_cap.mem_page_fault)) {
+        odp = UCT_IB_MLX5DV_ADDR_OF(query_hca_cap_out, out,
+                capability.odp_cap.memory_page_fault_scheme_cap);
+    } else {
+        if (md_config->devx_objs & UCS_BIT(UCT_IB_DEVX_OBJ_RCQP)) {
+            ucs_debug("%s: disable ODP because it's not supported for DEVX QP",
+                    uct_ib_device_name(&md->super.dev));
+            goto no_odp;
+        }
+
+        odp = UCT_IB_MLX5DV_ADDR_OF(query_hca_cap_out, out,
+                capability.odp_cap.transport_page_fault_scheme_cap);
```

Comment:
add ucs_debug print
ucs_debug("%s: memory page fault is supported", device_name)